config/k8s-etcd.txt

```diff
@@ -0,0 +1,116 @@
+# [This file is part of the zabbix-agent-osso package]
+k8s-etcd
+===========
+
+K8s-etcd monitors an in-cluster etcd setup.
+The status of the etcd is written to a configmap with the following(example)
+pod.
+
+statefulset:
+```
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  labels:
+    run: etcd-status-sender
+    application: patroni
+  name: etcd-status-sender
+  namespace: production-postgres
+spec:
+  selector:
+    matchLabels:
+      run: etcd-status-sender
+      application: patroni
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        run: etcd-status-sender
+        application: patroni
+    spec:
+      serviceAccountName: etcd-status-sender
+      containers:
+      - image: harbor.osso.io/ossobv/etcdctl:3.5.15
+        name: etcd-status-sender
+        env:
+        - name: ETCD_ENDPOINTS
+          value: patroni-etcd-0.patroni-etcd:2379,patroni-etcd-1.patroni-etcd:2379,patroni-etcd-2.patroni-etcd:2379
+        - name: POD_NAMESPACE
+          valueFrom:
+            fieldRef:
+              apiVersion: v1
+              fieldPath: metadata.namespace
+        - name: ETCD_CM_STATUS_NAME
+          value: etcd-status  # customize if needed
+        securityContext:
+          allowPrivilegeEscalation: false
+          capabilities:
+            drop:
+            - ALL
+          privileged: false
+          runAsNonRoot: true
+          seccompProfile:
+            type: RuntimeDefault
+        volumeMounts:
+        - mountPath: /tmp/scripts
+          name: etcd-status-script
+        command: ["/tmp/scripts/etcd-status.sh"]
+      volumes:
+      - name: etcd-status-script
+        configMap:
+          name: etcd-status-script
+          defaultMode: 0775
+      securityContext:
+        fsGroup: 1000
+        fsGroupChangePolicy: OnRootMismatch
+        runAsGroup: 1000
+        runAsNonRoot: true
+        runAsUser: 1000
+      dnsPolicy: ClusterFirst
+      restartPolicy: Always
+status: {}
+```
+script:
+```
+apiVersion: v1
+data:
+  etcd-status.sh: |
+    #!/bin/sh
+    CA_CERT="/var/run/secrets/kubernetes.io/serviceaccount/ca.crt";
+    sed -e 's/^/Authorization: Bearer /' /var/run/secrets/kubernetes.io/serviceaccount/token >/tmp/auth
+
+    while :
+    do
+      ETCD_STATUS="$(etcdctl endpoint status --endpoints=$ETCD_ENDPOINTS \
+        --write-out=json)"
+      echo "Etcd status: $ETCD_STATUS"
+      ETCD_HEALTH="$(etcdctl endpoint health --endpoints=$ETCD_ENDPOINTS \
+        --write-out=json)"
+      echo "Etcd health: $ETCD_HEALTH"
+      REQUEST_JSON=$(echo "{\"etcd_status\":${ETCD_STATUS},\"etcd_health\":${ETCD_HEALTH}}" | jq tostring)
+      PAYLOAD="[{\"op\":\"replace\",\"path\":\"/data/status\",\"value\":${REQUEST_JSON}}]"
+
+      curl --cacert ${CA_CERT} --header @/tmp/auth --header "Content-Type: \\
+        application/json-patch+json" --request PATCH --data "${PAYLOAD}" \
```

Comment:
Bad idea om daar midden in de string te breaken. Dit verandert de Content-Type string naar iets met extra whitespace. Gaat wel vaak goed (hier vast altijd), maar breek liever ergens met 0 change, bijvoorbeeld:
```
curl --cacert ${CA_CERT} --header @/tmp/auth \
    --header "Content-Type: application/json-patch+json" --request PATCH \
    --data "${PAYLOAD}" \
```